src/clj/fluree/db/json_ld/api.cljc

```diff
@@ -30,7 +31,9 @@
 
 (declare query)
 
-(defn promise-wrap
+(defn ^{:deprecated    "3.0"
```

Comment:
Do these deprecations issue any warnings to the user? And if so, is it at build time or run time?

Reply:
As far as I know, only dev time linters and IDE tools like clj-kondo and clojure-lsp (which uses clj-kondo) report on these metadata; there's no compiler option to print a warning as far as I can tell. We also should be able to make use of them in our docs generator to report on them as well (but I have to confirm this).

Do you think we need something more explicit?

Reply:
I think a `log/warn` message would be appropriate, something simple like `<fn> DEPRECATED. Superseded by <fn>`.